### PR TITLE
rpk: add --compression flag to rpk transform deploy

### PR DESCRIPTION
**Pull request by Deflaimun**
## Description

Resolves https://github.com/redpanda-data/documentation-private/issues/2583
Review deadline: Nov 4th (Monday)

## Page previews

https://deploy-preview-832--redpanda-docs-preview.netlify.app/current/reference/rpk/rpk-transform/rpk-transform-deploy/

## Checks

- [ ] New feature
- [x] Content gap
- [ ] Support Follow-up
- [ ] Small fix (typos, links, copyedits, etc)

### CODE REVIEW

**Review by oleiman — APPROVED** [review]
lgtm. thanks!

File: modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc [hunk 1]
```diff
@@ -34,8 +31,12 @@ rpk transform deploy [flags]
 |===
 |*Value* |*Type* |*Description*
 
+|--compression |string |Output batch compression type.
+
 |--file |string |The WebAssembly module to deploy.
 
+|--from-offset |string |Process an input topic partition from a relative offset; check help text for more information.
```
Comment (Deflaimun) [inline modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc]:
```suggestion
|--from-offset |string |Process an input topic partition from a relative offset. Run `rpk transform deploy --from-offset -h` for more information.

```

File: modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc [hunk 2]
```diff
@@ -35,7 +35,8 @@ rpk transform deploy [flags]
 
 |--file |string |The WebAssembly module to deploy.
 
-|--from-offset |string |Process an input topic partition from a relative offset; check help text for more information.
+|--from-offset |string |Process an input topic partition from a relative offset. Run `rpk transform deploy --from-offset -h` for more information.
```
Comment (oleiman) [inline modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc]: if this is the formulation we want, then it should be

```suggestion
|--from-offset |string |Process an input topic partition from a relative offset. Run `rpk transform deploy -h` for more information.
```

File: modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc [hunk 3]
```diff
@@ -34,8 +31,13 @@ rpk transform deploy [flags]
 |===
 |*Value* |*Type* |*Description*
 
+|--compression |string |Output batch compression type.
+
 |--file |string |The WebAssembly module to deploy.
 
+|--from-offset |string |Process an input topic partition from a relative offset. Run `rpk transform deploy --from-offset -h` for more information.
```
Comment (Deflaimun) [inline modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc]:
```suggestion
|--from-offset |string |Process an input topic partition from a relative offset.
```

File: modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc [hunk 4]
```diff
@@ -76,4 +77,24 @@ Deploy a transformation with multiple environment variables:
 rpk transform deploy --var FOO=BAR --var FIZZ=BUZZ
 ----
 
+Configure compression for batches output by data transforms. The default setting is "no compression," but you can choose from the following options:
```
Comment (kbatuigas) [inline modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc]: Is "no compression" the same as `none` listed below?

File: modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc [hunk 5]
```diff
@@ -76,4 +77,24 @@ Deploy a transformation with multiple environment variables:
...
+Configure this at deployment using `rpk` with the `--compression` flag:
+
+[,bash]
+----
+rpk transform deploy --compression <compression_type>
+----
+
+Enabling compression may increase computation costs and could impact latency at the output topic.
+
+For more details, see xref:modules:deploy/deployment-option/self-hosted/manual/sizing.adoc[].
```
Comment (kbatuigas) [inline modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc]:
```suggestion
For more details, see xref:deploy:deployment-option/self-hosted/manual/sizing.adoc[].
```
Please double check - maybe ROOT needs to be added?

**Review by kbatuigas — CHANGES_REQUESTED** [review]
One link to check

**Comment by Feediver1** [comment]
@Deflaimun What is the status on this one? Who might answer the question from @kbatuigas ?

File: modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc (hunk 5 above)
Comment (Deflaimun) [reply modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc]: That fix is correct. Thanks Kat

**Comment by Deflaimun** [comment]
@Feediver1 just need a confirmation from @oleiman.

File: modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc (hunk 4 above)
Comment (Deflaimun) [reply modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc]: Yes. I'll update

File: modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc [hunk 6]
```diff
@@ -76,4 +76,24 @@ Deploy a transformation with multiple environment variables:
 rpk transform deploy --var FOO=BAR --var FIZZ=BUZZ
 ----
 
+Configure compression for batches output by data transforms. The default setting is "no compression," but you can choose from the following options:
```
Comment (Deflaimun) [inline modules/reference/pages/rpk/rpk-transform/rpk-transform-deploy.adoc]:
```suggestion
Configure compression for batches output by data transforms. The default setting is `none` but you can choose from the following options:

```

**Comment by Deflaimun** [comment]
Ready for review. Confirmed that "no compression" is actually just the "none" value

**Review by oleiman — APPROVED** [review]
lgtm

**Review by kbatuigas — APPROVED** [review]

**Pull request merged**